endOfEpoch/metachain/trigger.go

```diff
@@ -53,53 +50,72 @@ func NewEndOfEpochTrigger(args *ArgsNewMetaEndOfEpochTrigger) (*trigger, error)
 	}
 
 	return &trigger{
-		rounder:                       args.Rounder,
 		roundsPerEpoch:                args.Settings.RoundsPerEpoch,
 		epochStartTime:                args.GenesisTime,
-		syncTimer:                     args.SyncTimer,
 		epoch:                         args.Epoch,
 		roundsBetweenForcedEndOfEpoch: args.Settings.MinRoundsBetweenEpochs,
+		rounder:                       args.Rounder,
 	}, nil
 }
 
 // IsEndOfEpoch return true if conditions are fullfilled for end of epoch
 func (t *trigger) IsEndOfEpoch() bool {
-	t.rounder.UpdateRound(t.epochStartTime, t.syncTimer.CurrentTime())
-	currRoundIndex := t.rounder.Index()
+	return t.isEndOfEpoch
+}
 
-	if currRoundIndex == 0 {
-		return true
+// ForceEndOfEpoch sets the conditions ofr end of epoch to true in case of edge cases
+func (t *trigger) ForceEndOfEpoch(round int64) error {
+	if t.currentRoundIndex > round {
+		return endOfEpoch.ErrSavedRoundIsHigherThanSaved
 	}
-
-	if currRoundIndex > t.roundsPerEpoch {
-		t.epoch += 1
-		t.epochStartTime = t.rounder.TimeStamp()
-		return true
+	if t.currentRoundIndex == round {
+		return endOfEpoch.ErrForceEndOfEpochCanBeCalledOnNewRound
+	}
+	if t.currentRoundIndex+1 != round {
```

Comment:
you can remove lines 68-73 as this condition covers both of them

Reply:
done